src/core/writer.js

```diff
@@ -0,0 +1,164 @@
+/* Copyright 2020 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint no-var: error */
+
+import { bytesToString, escapeString, stringToBytes } from "../shared/util.js";
+import { calculateMD5 } from "./crypto.js";
+import { Dict } from "./primitives.js";
```

Comment:
Note that these cannot be used on the main-thread, since that will end up pulling in *a lot* of the `src/core/` functionality into the *built* `pdf.js` file. (The referenced code should only appear in the *built* `pdf.worker.js` file.)

It may be helpful to look in the `gulpfile.js` or perhaps simply building the generic PDF.js library, using `gulp generic`, and look at the where various code ends up (with/without these patches applied).